src/main/java/ro/msg/learning/shop/services/ReveneuServices.java

```diff
@@ -0,0 +1,64 @@
+package ro.msg.learning.shop.services;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.convert.ReveneuConverter;
+import ro.msg.learning.shop.dto.ReveneuDTO;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.Ordeer;
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Revenue;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.OrderDetailRepository;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ReveneuRepository;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ReveneuServices {
+    private final LocationRepository locationRepository;
+    private final OrderRepository orderRepository;
+    private final OrderDetailRepository orderDetailRepository;
+    private final ReveneuRepository reveneuRepository;
+    private final ReveneuConverter reveneuConverter;
+
+    @Scheduled(cron = "0 59 23 * * ?")
+    public void aggregationSum() {
+        List<Location> locations = locationRepository.findAll();
+        LocalDate date = LocalDate.now().minusDays(1); //yesterday
+        locations.forEach(location -> {
```

Comment:
It is bad practice to do save inside a loop. Create a list of revenue objects and then save all of them.